### ISSUE TITLE
Fix S4545FP: Static members in same namespace or FQ members can be called

### ISSUE THREAD

**Issue by manfred-brands** [SonarSource/sonar-dotnet#9770]
### Description

We have an Angle class which holds data in Radians an an extension method for display purposes to convert its value to degrees.

`[DebuggerDisplay("Degrees {ToDecimalDegreesExtensions.ToDegrees(this)}; Radians {Radians}")]`

This is flagged with `ToDecimalDegreesExtension` doesn't exist.

<img width="788" height="149" alt="Image" src="https://github.com/user-attachments/assets/936e4bae-93dc-45b8-8f9e-edc6f14358f6" />

With a debug display showing the proper value:

<img width="603" height="56" alt="Image" src="https://github.com/user-attachments/assets/c7f337ff-bfc9-4cf9-834f-dcab0d4fe5a0" />

### Reproducer

```csharp
#pragma warning disable IDE0065 // Misplaced using directive
#pragma warning disable SA1403 // File may only contain a single namespace
#pragma warning disable SA1649 // File name should match first type name

using System;
using System.Diagnostics;

namespace SonarSource.Issues
{
#pragma warning disable S4545 // "DebuggerDisplayAttribute" strings should reference existing members
    [DebuggerDisplay("Degrees {ToDecimalDegreesExtensions.ToDegrees(this)}; Radians {Radians}")]
#pragma warning restore S4545 // "DebuggerDisplayAttribute" strings should reference existing members
    internal readonly record struct Angle(double Radians)
    {
        public static Angle Degree180 { get; } = new Angle(Math.PI);
    }

    internal static class ToDecimalDegreesExtensions
    {
        public static double ToDegrees(this Angle angle)
        {
            return 180.0 / Math.PI * angle.Radians;
        }
    }

    namespace NUnit
    {
        using global::NUnit.Framework;

        [TestFixture]
        internal sealed class AngleFixture
        {
            [Test]
            public void Degree180()
            {
                Assert.That(Angle.Degree180.Radians, Is.EqualTo(Math.PI));
            }
        }
    }
}
```

### Product and Version

NuGet SonarAnalyzer.CSharp version 10.15.0.120848

**Comment by alex-meseldzija-sonarsource** [comment]
Hey, 

Closing as a duplicate of [NET-2620](https://sonarsource.atlassian.net/browse/NET-2620) (internal only).

Just a note you can workaround the issue in this code using `global::SonarSource.Issues.ToDecimalDegreesExtensions.ToDegrees(this)`

[NET-2620]: https://sonarsource.atlassian.net/browse/NET-2620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by manfred-brands** [comment]
@alex-meseldzija-sonarsource You marked this as closed, but the latest 10.17 release does not contain a fix for this.
Please only close items after they are fixed as we have no access to your internal JIRA tickets.

**Comment by alex-meseldzija-sonarsource** [comment]
Hi @manfred-brands,
Thanks for the feedback, I'll discuss with the team to find a better way to handle duplicates of internal issues.

On a related note, we just released version [10.18](https://github.com/SonarSource/sonar-dotnet/releases/tag/10.18.0.131500) of our analyzers which fixes this issue.
